Signals not getting caught on Linux

It looks like it's catching the signal, but then failing to un-hook itself and re-kill or something?

```
> signal-exit@2.0.0 test /home/isaacs/signal-exit
> standard && nyc tap --timeout=240 ./test/*.js

./test/all-integration-test.js ...................... 31/56
  all-signals-integration-test > exits properly: SIGABRT
  not ok AssertionError: null == true
    at:
      file: test/all-integration-test.js
      line: 43
      column: 38
    name: AssertionError
    actual: null
    expected: true
    operator: ==
    generatedMessage: true
    test: 'exits properly: SIGABRT'
    message: 'AssertionError: null == true'
    source: |
      if (!process.env.TRAVIS) assert(err.signal)
    stack: |
      test/all-integration-test.js:43:38
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGALRM
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGALRM'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGBUS
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGBUS'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGFPE
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGFPE'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGHUP
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGHUP'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGILL
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGILL'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGINT
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGINT'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGIOT
  not ok AssertionError: null == true
    at:
      file: test/all-integration-test.js
      line: 43
      column: 38
    name: AssertionError
    actual: null
    expected: true
    operator: ==
    generatedMessage: true
    test: 'exits properly: SIGIOT'
    message: 'AssertionError: null == true'
    source: |
      if (!process.env.TRAVIS) assert(err.signal)
    stack: |
      test/all-integration-test.js:43:38
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGPIPE
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGPIPE'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGPROF
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGPROF'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGQUIT
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGQUIT'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGSEGV
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGSEGV'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGSYS
  not ok AssertionError: null == true
    at:
      file: test/all-integration-test.js
      line: 43
      column: 38
    name: AssertionError
    actual: null
    expected: true
    operator: ==
    generatedMessage: true
    test: 'exits properly: SIGSYS'
    message: 'AssertionError: null == true'
    source: |
      if (!process.env.TRAVIS) assert(err.signal)
    stack: |
      test/all-integration-test.js:43:38
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGTERM
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGTERM'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGTRAP
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGTRAP'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGUSR1
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGUSR1'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGUSR2
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGUSR2'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGVTALRM
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGVTALRM'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGXCPU
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGXCPU'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGXFSZ
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGXFSZ'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGIO
  not ok AssertionError: null == true
    at:
      file: test/all-integration-test.js
      line: 43
      column: 38
    name: AssertionError
    actual: null
    expected: true
    operator: ==
    generatedMessage: true
    test: 'exits properly: SIGIO'
    message: 'AssertionError: null == true'
    source: |
      if (!process.env.TRAVIS) assert(err.signal)
    stack: |
      test/all-integration-test.js:43:38
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGPOLL
  not ok AssertionError: null == true
    at:
      file: test/all-integration-test.js
      line: 43
      column: 38
    name: AssertionError
    actual: null
    expected: true
    operator: ==
    generatedMessage: true
    test: 'exits properly: SIGPOLL'
    message: 'AssertionError: null == true'
    source: |
      if (!process.env.TRAVIS) assert(err.signal)
    stack: |
      test/all-integration-test.js:43:38
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGPWR
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGPWR'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGSTKFLT
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/all-integration-test.js
      line: 41
      column: 48
    test: 'exits properly: SIGSTKFLT'
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal(sig)
    stack: |
      test/all-integration-test.js:41:48
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  all-signals-integration-test > exits properly: SIGUNUSED
  not ok AssertionError: null == true
    at:
      file: test/all-integration-test.js
      line: 43
      column: 38
    name: AssertionError
    actual: null
    expected: true
    operator: ==
    generatedMessage: true
    test: 'exits properly: SIGUNUSED'
    message: 'AssertionError: null == true'
    source: |
      if (!process.env.TRAVIS) assert(err.signal)
    stack: |
      test/all-integration-test.js:43:38
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

./test/signal-exit-test.js .......................... 10/11
  signal-exit > removes handlers when fully unwrapped
  not ok TypeError: Cannot read property 'should' of null
    at:
      file: test/signal-exit-test.js
      line: 89
      column: 42
    test: removes handlers when fully unwrapped
    message: "TypeError: Cannot read property 'should' of null"
    source: |
      if (!process.env.TRAVIS) err.signal.should.equal('SIGHUP')
    stack: |
      test/signal-exit-test.js:89:42
      ChildProcess.exithandler (child_process.js:220:5)
      maybeClose (internal/child_process.js:818:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

total ............................................... 41/67


  41 passing (15s)
  26 failing

npm ERR! Test failed.  See above for more details.
```

--- a/.gitignore
+++ b/.gitignore
@@ -2,3 +2,4 @@ node_modules
 .DS_Store
 nyc_output
 coverage
+.nyc_output

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,6 @@
 sudo: false
 language: node_js
 node_js:
-- '0.12'
-- '0.10'
-- iojs
-after_success: npm run coverage
+  - '0.12'
+  - '4'
+  - '5'

--- a/package.json
+++ b/package.json
@@ -4,7 +4,8 @@
   "description": "when you want to fire an event no matter how a process exits.",
   "main": "index.js",
   "scripts": {
-    "test": "standard && nyc tap --timeout=240 ./test/*.js",
+    "posttest": "standard",
+    "test": "tap --timeout=240 ./test/*.js --cov",
     "coverage": "nyc report --reporter=text-lcov | coveralls"
   },
   "repository": {

--- a/package.json
+++ b/package.json
@@ -25,8 +25,8 @@
   "devDependencies": {
     "chai": "^2.3.0",
     "coveralls": "^2.11.2",
-    "nyc": "^2.1.2",
+    "nyc": "^5.1.0",
     "standard": "^3.9.0",
-    "tap": "1.0.4"
+    "tap": "^2.3.5"
   }
 }

--- a/test/all-integration-test.js
+++ b/test/all-integration-test.js
@@ -1,7 +1,8 @@
 /* global describe, it */
 
 var exec = require('child_process').exec,
-  assert = require('assert')
+  assert = require('assert'),
+  shell = process.platform === 'win32' ? null : { shell: '/bin/bash' }
 
 require('chai').should()
 require('tap').mochaGlobals()
@@ -32,7 +33,8 @@ describe('all-signals-integration-test', function () {
       // travis has issues with SIGUSR1 on Node 0.x.10.
       if (process.env.TRAVIS && sig === 'SIGUSR1') return done()
 
-      exec(node + ' ' + js + ' ' + sig, function (err, stdout, stderr) {
+      var cmd = node + ' ' + js + ' ' + sig
+      exec(cmd, shell, function (err, stdout, stderr) {
         if (sig) {
           assert(err)
           if (!isNaN(sig)) {
@@ -71,7 +73,7 @@ describe('all-signals-integration-test', function () {
       if (process.env.TRAVIS && sig === 'SIGUSR1') return done()
 
       var cmd = node + ' ' + js + ' ' + sig
-      exec(cmd, function (err, stdout, stderr) {
+      exec(cmd, shell, function (err, stdout, stderr) {
         assert.ifError(err)
         try {
           var data = JSON.parse(stdout)

--- a/test/all-integration-test.js
+++ b/test/all-integration-test.js
@@ -30,8 +30,8 @@ describe('all-signals-integration-test', function () {
     var node = process.execPath
     var js = require.resolve('./fixtures/exiter.js')
     it('exits properly: ' + sig, function (done) {
-      // travis has issues with SIGUSR1 on Node 0.x.10.
-      if (process.env.TRAVIS && sig === 'SIGUSR1') return done()
+      // issues with SIGUSR1 on Node 0.10.x
+      if (process.version.match(/^v0\.10\./) && sig === 'SIGUSR1') return done()
 
       var cmd = node + ' ' + js + ' ' + sig
       exec(cmd, shell, function (err, stdout, stderr) {
@@ -40,9 +40,9 @@ describe('all-signals-integration-test', function () {
           if (!isNaN(sig)) {
             assert.equal(err.code, sig)
           } else if (!weirdSignal(sig)) {
-            if (!process.env.TRAVIS) err.signal.should.equal(sig)
+            err.signal.should.equal(sig)
           } else if (sig) {
-            if (!process.env.TRAVIS) assert(err.signal)
+            assert(err.signal)
           }
         } else {
           assert.ifError(err)
@@ -69,8 +69,8 @@ describe('all-signals-integration-test', function () {
     var node = process.execPath
     var js = require.resolve('./fixtures/parent.js')
     it('exits properly: (external sig) ' + sig, function (done) {
-      // travis has issues with SIGUSR1 on Node 0.x.10.
-      if (process.env.TRAVIS && sig === 'SIGUSR1') return done()
+      // issues with SIGUSR1 on Node 0.10.x
+      if (process.version.match(/^v0\.10\./) && sig === 'SIGUSR1') return done()
 
       var cmd = node + ' ' + js + ' ' + sig
       exec(cmd, shell, function (err, stdout, stderr) {

--- a/test/fixtures/change-code.js
+++ b/test/fixtures/change-code.js
@@ -76,7 +76,8 @@ function listener (code, signal) {
 
 function run (opt) {
   console.error(opt)
-  exec(process.execPath + ' ' + __filename + ' ' + opt, function (err, stdout, stderr) {
+  var shell = process.platform === 'win32' ? null : { shell: '/bin/bash' }
+  exec(process.execPath + ' ' + __filename + ' ' + opt, shell, function (err, stdout, stderr) {
     var res = JSON.parse(stdout)
     if (err) {
       res.actualCode = err.code

--- a/test/multi-exit.js
+++ b/test/multi-exit.js
@@ -1,5 +1,6 @@
 var exec = require('child_process').exec,
-  t = require('tap')
+  t = require('tap'),
+  shell = process.platform === 'win32' ? null : { shell: '/bin/bash' }
 
 var fixture = require.resolve('./fixtures/change-code.js')
 var expect = require('./fixtures/change-code-expect.json')
@@ -41,7 +42,7 @@ types.forEach(function (type) {
 opts.forEach(function (opt) {
   t.test(opt, function (t) {
     var cmd = process.execPath + ' ' + fixture + ' ' + opt
-    exec(cmd, function (err, stdout, stderr) {
+    exec(cmd, shell, function (err, stdout, stderr) {
       var res = JSON.parse(stdout)
       if (err) {
         res.actualCode = err.code

--- a/test/signal-exit-test.js
+++ b/test/signal-exit-test.js
@@ -2,7 +2,8 @@
 
 var exec = require('child_process').exec,
   expect = require('chai').expect,
-  assert = require('assert')
+  assert = require('assert'),
+  shell = process.platform === 'win32' ? null : { shell: '/bin/bash' }
 
 require('chai').should()
 require('tap').mochaGlobals()
@@ -10,7 +11,7 @@ require('tap').mochaGlobals()
 describe('signal-exit', function () {
 
   it('receives an exit event when a process exits normally', function (done) {
-    exec(process.execPath + ' ./test/fixtures/end-of-execution.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/end-of-execution.js', shell, function (err, stdout, stderr) {
       expect(err).to.equal(null)
       stdout.should.match(/reached end of execution, 0, null/)
       done()
@@ -18,7 +19,7 @@ describe('signal-exit', function () {
   })
 
   it('receives an exit event when a process is terminated with sigint', function (done) {
-    exec(process.execPath + ' ./test/fixtures/sigint.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/sigint.js', shell, function (err, stdout, stderr) {
       assert(err)
       stdout.should.match(/exited with sigint, null, SIGINT/)
       done()
@@ -26,7 +27,7 @@ describe('signal-exit', function () {
   })
 
   it('receives an exit event when a process is terminated with sigterm', function (done) {
-    exec(process.execPath + ' ./test/fixtures/sigterm.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/sigterm.js', shell, function (err, stdout, stderr) {
       assert(err)
       stdout.should.match(/exited with sigterm, null, SIGTERM/)
       done()
@@ -34,7 +35,7 @@ describe('signal-exit', function () {
   })
 
   it('receives an exit event when process.exit() is called', function (done) {
-    exec(process.execPath + ' ./test/fixtures/exit.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/exit.js', shell, function (err, stdout, stderr) {
       err.code.should.equal(32)
       stdout.should.match(/exited with process\.exit\(\), 32, null/)
       done()
@@ -42,7 +43,7 @@ describe('signal-exit', function () {
   })
 
   it('does not exit if user handles signal', function (done) {
-    exec(process.execPath + ' ./test/fixtures/signal-listener.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/signal-listener.js', shell, function (err, stdout, stderr) {
       assert(err)
       assert.equal(stdout, 'exited calledListener=4, code=null, signal="SIGHUP"\n')
       done()
@@ -50,7 +51,7 @@ describe('signal-exit', function () {
   })
 
   it('ensures that if alwaysLast=true, the handler is run last (signal)', function (done) {
-    exec(process.execPath + ' ./test/fixtures/signal-last.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/signal-last.js', shell, function (err, stdout, stderr) {
       assert(err)
       stdout.should.match(/first counter=1/)
       stdout.should.match(/last counter=2/)
@@ -59,7 +60,7 @@ describe('signal-exit', function () {
   })
 
   it('ensures that if alwaysLast=true, the handler is run last (normal exit)', function (done) {
-    exec(process.execPath + ' ./test/fixtures/exit-last.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/exit-last.js', shell, function (err, stdout, stderr) {
       assert.ifError(err)
       stdout.should.match(/first counter=1/)
       stdout.should.match(/last counter=2/)
@@ -68,7 +69,7 @@ describe('signal-exit', function () {
   })
 
   it('works when loaded multiple times', function (done) {
-    exec(process.execPath + ' ./test/fixtures/multiple-load.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/multiple-load.js', shell, function (err, stdout, stderr) {
       assert(err)
       stdout.should.match(/first counter=1, code=null, signal="SIGHUP"/)
       stdout.should.match(/first counter=2, code=null, signal="SIGHUP"/)
@@ -80,7 +81,7 @@ describe('signal-exit', function () {
 
   // TODO: test on a few non-OSX machines.
   it('removes handlers when fully unwrapped', function (done) {
-    exec(process.execPath + ' ./test/fixtures/unwrap.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/unwrap.js', shell, function (err, stdout, stderr) {
       // on Travis CI no err.signal is populated but
       // err.code is 129 (which I think tends to be SIGHUP).
       var expectedCode = process.env.TRAVIS ? 129 : null
@@ -93,14 +94,14 @@ describe('signal-exit', function () {
   })
 
   it('does not load() or unload() more than once', function (done) {
-    exec(process.execPath + ' ./test/fixtures/load-unload.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/load-unload.js', shell, function (err, stdout, stderr) {
       assert.ifError(err)
       done()
     })
   })
 
   it('handles uncatchable signals with grace and poise', function (done) {
-    exec(process.execPath + ' ./test/fixtures/sigkill.js', function (err, stdout, stderr) {
+    exec(process.execPath + ' ./test/fixtures/sigkill.js', shell, function (err, stdout, stderr) {
       assert.ifError(err)
       done()
     })

--- a/test/signal-exit-test.js
+++ b/test/signal-exit-test.js
@@ -82,13 +82,9 @@ describe('signal-exit', function () {
   // TODO: test on a few non-OSX machines.
   it('removes handlers when fully unwrapped', function (done) {
     exec(process.execPath + ' ./test/fixtures/unwrap.js', shell, function (err, stdout, stderr) {
-      // on Travis CI no err.signal is populated but
-      // err.code is 129 (which I think tends to be SIGHUP).
-      var expectedCode = process.env.TRAVIS ? 129 : null
-
       assert(err)
-      if (!process.env.TRAVIS) err.signal.should.equal('SIGHUP')
-      expect(err.code).to.equal(expectedCode)
+      err.signal.should.equal('SIGHUP')
+      expect(err.code).to.equal(null)
       done()
     })
   })

